### ISSUE TITLE
Support for returning the result as a JSON String

### ISSUE THREAD

**Issue by hardikSinghBehl** [instancio/instancio#1048]
It'll be cool to have a `.toJson()` method that converts the generated test data into a JSON string: 

```java
String request = Instancio.of(UserCreationRequestDto.classs).toJson();
```

**Comment by armandino** [comment]
Thank you for the suggestion! I did consider it at one point, but decided against it as I didn't want to introduce an extra dependency.

One possibility is to introduce a new API that maps the result to another type using a `Function<T,R>`-like interface, e.g.

```java
String request = Instancio.of(UserCreationRequestDto.classs).as(json());
```

where the `json()` is simple a static utility method (defined by the user) that converts an object to JSON. This approach should allow converting the object to other formats/types. What do you think?

**Comment by armandino** [comment]
@hardikSinghBehl thanks for the additional feedback. At this point I'm not sure about committing to a JSON library and adding a `toJson()` method. I'm also not sure if an optional dependency is the best approach (e.g. which library should it be? what about users who use a different library?). I'll need to think about it some more. I'll keep this issue open to see how much demand there is for the `toJson()` method and to collect more ideas.

I'll add the `as(Function)` method. Hopefully that works for your use case for the time being.

**Comment by hardikSinghBehl** [comment]
sure @armandino, makes sense. Thank you for the prompt action!

**Comment by armandino** [comment]
@hardikSinghBehl just released a new version `4.8.0` with the `as(Function)` method. Let me know if you run into any issues.

**Comment by hardikSinghBehl** [comment]
Thanks for the update @armandino. Tried the new method, it's working as expected!
